addons/host-upgrades/addon.rb

```diff
@@ -5,16 +5,48 @@
   license 'Apache License 2.0'
 
   config {
-    attribute :interval, Pharos::Types::String
+    attribute :schedule, Pharos::Types::String
+    attribute :schedule_window, Pharos::Types::String
+    attribute :reboot, Pharos::Types::Bool.default(false)
+    attribute :drain, Pharos::Types::Bool.default(true)
   }
 
   config_schema {
-    required(:interval).filled(:str?, :duration?)
+    configure do
+      SCHEDULE_RE = %r(^([*?]|[A-Z0-9/,-]+)(\s+([*?]|[A-Z0-9/,-]+)){5}|@\w+$)
+      DURATION_RE = /^(\d+(\.\d+)?(ns|us|ms|s|m|h|))+$/
+
+      def schedule?(value)
+        !!SCHEDULE_RE.match?(value)
```

Comment:
Just wondering should we use valid cron schedule (we have fugit gem as a dep)?

Reply:
The `pharos-cluster` itself doesn't need to actually parse the cron string, but `Fugit::Cron.parse` would probably be better than a regexp... unfortunately, the github.com/robfig/cron package used by `pharos-host-upgrades` doesn't accept the standard five-field cron syntax, it expects an extended cron syntax with a leading seconds field... which is a bad idea IMO, it would be better if this used a standard cron syntax, we don't really need per-second granularity here.

Maybe we should change `pharos-host-upgrades` to accept a standard five-field cron syntax before releasing this?

Reply:
I meant that we could use `Fugit::Cron.parse` to parse standard five-field cron and then addon could convert it to something that `pharos-host-upgrades` understands.

Reply:
I think that conversion should happen within `pharos-host-upgrades`.

Reply:
Looks like `Fugit::Cron.parse` also supports the alternative six-field cron syntax, which confuses things a bit now...

```
irb(main):003:0> Fugit::Cron.parse('0 0 0 * * *')
=> #<Fugit::Cron:0x0000000001c172c0 @original="0 0 0 * * *", @cron_s=nil, @seconds=[0], @minutes=[0], @hours=[0], @monthdays=nil, @months=nil, @weekdays=nil, @zone=nil, @timezone=nil>
```